app/controllers/doc_controller.rb

```diff
@@ -88,6 +89,10 @@ def ext
 
   private
 
+  def set_caching
+    expires_in 10.minutes, :public => true
+  end
+
```

Comment:
You don't want to enable caching e.g. for search controller as each time you query it, the result can be different. The two types of pages on git-scm that don't change are from site_controller and doc_controller, and that's where I enabled caching.